modules/helper/EsxiServiceHelper.py

```diff
@@ -0,0 +1,106 @@
+from interfaces.host import Host
+from modules.api.Atlas import Atlas
+from modules.api.SshHelper import SshHelper
+
+from threading import Thread
+from queue import Queue
+from typing import List
+
+import logging
+
+logger = logging.getLogger('esxi')
+
+
+class EsxiServiceHelper:
+
+    def __init__(self, atlas: Atlas, esxi_username: str, esxi_password: str, monitored_services, max_threads: int):
+        """
+        Helper providing functionality to query critical service stats of esxi-host
+        """
+
+        self.atlas = atlas
+        self.esxi_username = esxi_username
+        self.esxi_password = esxi_password
+        self.max_threads = max_threads
+
+        self._services = monitored_services
+        command_list = ["/etc/init.d/%s status" %
+                        service for service in self._services]
+        self._command = ' & '.join(command_list)
+
+        # Strange bug. Display name is different from registered service name
+        self._services = [item.replace('nsx-opsagent', 'opsAgent')
+                          for item in self._services]
+
+    def _worker(host_queue: Queue, ssh_username: str, ssh_password: str,
+                command: str, services: list, output: list):
+        """
+        Multithreaded worker. Used for SSH-Connections to esxi-hosts
+        """
+
+        while not host_queue.empty():
+            host: Host = host_queue.get()
+
+            answer = SshHelper.execute_command(
+                host.address,
+                ssh_username,
+                ssh_password,
+                command
+            )
+
+            host.services = {}
+            if answer is None:
+                continue
```

Comment:
maybe some info logging here, that we are having an empty answer for host XY

Reply:
```python
logger.error('critical_service_collector: Could not fetch any services for host: %s' % host.name)
```